### PR TITLE
Claim & Reserve Task for Exploit & Reserve Flags

### PR DESCRIPTION
**Pull request by Dasku**
Claiming and reserving now working as intended. 

They spawn from exploit and reserve flags but the code is commented for mining flag too when its ready or for people to try them both in combination. 

Remaining issue:
Check controller ticks for < 2000

### CODE REVIEW

File: task.claim.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        // a creep completed spawning
+        Creep.spawningCompleted.on( creep => Task.claim.handleSpawningCompleted(creep) );
+        // a creep will die soon
+        Creep.predictedRenewal.on( creep => Task.claim.handleCreepDied(creep.name) );
+        // a creep died
+        Creep.died.on( name => Task.claim.handleCreepDied(name) );
+    },
+    // for each flag
+    handleFlagFound: flag => {
+        // if it is a yellow/yellow flag
```
Comment (UntilSunrise) [inline task.claim.js]: Claim Flag is now green/green

File: task.claim.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        let memory = Task.claim.memory(flag);
+        // count creeps assigned to task
+        let count = memory.queued.length + memory.spawning.length + memory.running.length;
+        // if creep count below requirement spawn a new creep creep
+        if( count < 1 ) {
+            // get nearest room
+            let room = Room.bestSpawnRoomFor(flag.pos.roomName);
+            // define new creep
+            let fixedBody = [CLAIM, MOVE];
+            let multiBody = [];
```
Comment (UntilSunrise) [inline task.claim.js]: The creep will not be strong enough if the controller is still controlled by an enemy (http://support.screeps.com/hc/en-us/articles/203013212-Creep#attackController)

**Comment by Dasku** [comment]
I'm working on a new action for attackController separately :)

File: task.claim.js (hunk 1 above)
Comment (Dasku) [inline task.claim.js]: Yep, Ill update that comment, thanks.

File: task.claim.js (hunk 2 above)
Comment (Dasku) [inline task.claim.js]: Hi UntilSunrise,

You are correct, and this was intentional. I will be added a separate attackController task to specifically generate claim creeps big enough for this task.

File: task.reserve.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            // flag.color == FLAG_COLOR.claim.mining.color && flag.secondaryColor == FLAG_COLOR.claim.mining.secondaryColor
+
+            // check if a new creep has to be spawned
+            
+            Task.reserve.checkForRequiredCreeps(flag);
+        }
+    },
+    // check if a new creep has to be spawned
+    checkForRequiredCreeps: (flag) => {
+        // get task memory
```
Comment (cyberblast) [inline task.reserve.js]: if( flag && flag.room && flag.room.controller && flag.room.controller.reservation && flag.room.controller.reservation.ticksToEnd > 2500)
                                    return;

Comment (Dasku) [inline task.reserve.js]: Done, thanks

**Comment by webremote** [comment]
`[11:25:14 PM]E66N79 > Queued creep too big for room: {"parts":["claim","claim","move","move"],"name":"reserve-E67N79","behaviour":"claimer","setup":"claimer","destiny":{"task":"reserve","flagName":"Flag11"}}` after update

**Comment by Dasku** [comment]
Thanks, I've just submitted a fix for that.

**Pull request merged**